isisd/isis_bfd.c

```diff
@@ -167,11 +167,15 @@ void isis_bfd_circuit_cmd(struct isis_circuit *circuit)
 
 			struct listnode *node;
 			struct isis_adjacency *adj;
+			struct list *adj_list;
 
 			if (!adjdb)
 				continue;
-			for (ALL_LIST_ELEMENTS_RO(adjdb, node, adj))
+			adj_list = list_new();
+			isis_adj_build_up_list(circuit->u.bc.adjdb[level - 1], adj_list);
+			for (ALL_LIST_ELEMENTS_RO(adj_list, node, adj))
```

Comment:
I don't understand the crash explanation in the commit message. `isis_bfd.c` has only 233 LOC but your report says the assert happens on line 269. Something seems wrong.